pkg/kapp/resources/resources.go

```diff
@@ -472,12 +475,20 @@ func (c *ResourcesImpl) resourceErr(err error, action string, resource Resource)
 	return resourcePlainErr{err, action, resource}
 }
 
-func (c *ResourcesImpl) resourceClient(resource Resource) (dynamic.ResourceInterface, ResourceType, error) {
+type resourceClientOpts struct {
+	warnings bool
+}
+
+func (c *ResourcesImpl) resourceClient(resource Resource, opts resourceClientOpts) (dynamic.ResourceInterface, ResourceType, error) {
 	resType, err := c.resourceTypes.Find(resource)
 	if err != nil {
 		return nil, ResourceType{}, err
 	}
 
+	if !opts.warnings {
+		return c.mutedDynamicClient.Resource(resType.GroupVersionResource).Namespace(resource.Namespace()), resType, nil
```

Comment:
instead of doing the same line below, lets have a variable dynamicClient which is set to mutedDynamicClient and if !opts.Warnings then it gets set to dynamicClient. makes it easier to tell that same logic is applied on both clients.

Reply:
Improved this conditional!